### PR TITLE
Creating specific datasets for scenes

### PR DESCRIPTION
**Pull request by pfmark**
# Summary
We'll treat datasets that contain scenes differently in the frontend (additional tab) compared to those without scenes. Therefore we need to introduce an attribute for a dataset which specifies the type of dataset as a boolean (`is_scene`). This attribute needs to be set during dataset creation. 

Regarding the minor version, I assumed, we need to see if we merge [Gunnar's PR](https://github.com/scaleapi/nucleus-python-client/pull/159) first. So if this one would be merged first, it should be `0.5.0` instead. 

# How to test
This PR depends on the backend changes introduced in PR [#33106](https://github.com/scaleapi/scaleapi/pull/32750). To test it, do the following
- set `NUCLEUS_ENDPOINT = "http://localhost:3000/v1/nucleus"` in `nucleus/constants.py`
- Install this client from `pip install .` from source folder
- Check out `scaleapi` branch `mark-add-scene-type`
- yarn `dev-prodclone-backend`
- yarn `dev-prodclone-sfn`
- Create a dataset and upload a scene through the API. I can share a little script for that, let me know if you need sth.

### CODE REVIEW

File: tests/test_dataset.py [hunk 1]
```diff
@@ -99,11 +99,28 @@ def make_dataset_items():
     return ds_items_with_metadata
 
 
-def test_dataset_create_and_delete(CLIENT):
+def test_dataset_create_and_delete_no_scene(CLIENT):
```
Comment (ardila) [inline tests/test_dataset.py:102]: Shouldn't you also have tests for what effect this has on upload?

**Comment by ardila** [comment]
Why are we making users explicitly set this rather than just inferring it from the data they send in?

**Comment by ardila** [comment]
I'll LG for now since I assume these design discussions have alreaady ocurred and I don't want to be an obstacle to progress here. We do need to immediately update the guides for scene uploads concurrently with the release of any version of the client which includes this change.

File: nucleus/__init__.py [hunk 2]
```diff
@@ -337,6 +339,7 @@ def create_dataset(
 
         Parameters:
             name: A human-readable name for the dataset.
+            is_scene: Boolean specifying if the dataset contains / will contain scenes. Cannot be changed at a later point in time.
```
Comment (ardila) [inline nucleus/__init__.py:342]: Add more detail on what this will affect.

**Review by ardila — APPROVED** [review]

**Comment by pfmark** [comment]
> I'll LG for now since I assume these design discussions have alreaady ocurred and I don't want to be an obstacle to progress here. We do need to immediately update the guides for scene uploads concurrently with the release of any version of the client which includes this change.

Yup, I had it implicitly in the beginning but @song-william saw DB issues there. Therefore it's explicit now. I'm also fine having the default value `True` now. This will soon end up in an additional empty tab in the frontend then though.

**Comment by jean-lucas** [comment]
I also agree that the `is_scene` should be set programmatically, without the user input. Two reasons i can think of why:
- will break current workflows
- user may not know what a scene is

**Comment by pfmark** [comment]
> I also agree that the is_scene should be set programmatically, without the user input. Two reasons i can think of why:

See discussion [here](https://github.com/scaleapi/scaleapi/pull/33106), it shouldn't change for a dataset.

> will break current workflows

Not really, we just need to update the docs accordingly. 

> user may not know what a scene is

I'm pretty sure they wouldn't upload a scene in that case. 

As I mentioned above: 
- Either we set the default value to `False` + careful update of the docs (preferred)
- Set default to `True` and live with the empty scene tab in the frontend

File: nucleus/dataset.py [hunk 3]
```diff
@@ -118,6 +118,14 @@ def name(self) -> str:
             )["name"]
         return self._name
 
+    @property
+    def is_scene(self) -> bool:
```
Comment (jean-lucas) [inline nucleus/dataset.py:122]:
```
dataset = client.create_dataset('Test PR 165', is_scene=True)
dataset.is_scene
```

Gives the error:
```
NucleusAPIError: Your client is on version 0.3.1. If you have not recently done so, please make sure you have updated to the latest version of the client by running pip install --upgrade scale-nucleus
Tried to get http://localhost:3000/v1/nucleus/dataset/ds_c6n3ysk5e8cp828vqnj0/is_scene, but received 404: Not Found.
The detailed error is:
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /v1/nucleus/dataset/ds_c6n3ysk5e8cp828vqnj0/is_scene</pre>
</body>
</html>
```

File: nucleus/dataset.py [hunk 4]
```diff
@@ -118,6 +118,14 @@ def name(self) -> str:
...
+        """If the dataset can contain scenes or not."""
+        response = self._client.make_request(
+            {}, f"dataset/{self.id}/is_scene", requests.get
+        )["is_scene"]
```
Comment (jean-lucas) [inline nucleus/dataset.py:126]:
```suggestion
        )[DATASET_IS_SCENE_KEY]
```

**Review by jean-lucas — CHANGES_REQUESTED** [review]
You may also need to update the patch or minor version of the pip package before merging

File: nucleus/dataset.py (hunk 4 above)
Comment (jean-lucas) [reply nucleus/dataset.py:126]: Maybe need to add the import. Haven't tested suggestion

File: nucleus/dataset.py (hunk 3 above)
Comment (jean-lucas) [reply nucleus/dataset.py:122]: it works now after pulling

File: tests/test_dataset.py (hunk 1 above)
Comment (pfmark) [reply tests/test_dataset.py:102]: I'm testing this properly in the backend already. You think I should test it here again?

File: nucleus/dataset.py (hunk 4 above)
Comment (jean-lucas) [reply nucleus/dataset.py]:
```
/usr/local/lib/python3.9/site-packages/nucleus/dataset.py in is_scene(self)
    124         response = self._client.make_request(
    125             {}, f"dataset/{self.id}/is_scene", requests.get
--> 126         )[DATASET_IS_SCENE_KEY]
    127         return response
    128 

NameError: name 'DATASET_IS_SCENE_KEY' is not defined
```

File: nucleus/__init__.py [hunk 5]
```diff
@@ -329,6 +330,7 @@ def create_dataset_from_project(
     def create_dataset(
```
[the remark this answers is not in the document]
Comment (pfmark) [reply nucleus/__init__.py:330]: Should be resolved now, right?

[the remark this answers is not in the document]
Comment (pfmark) [reply nucleus/__init__.py:330]: Okay will do

[the remark this answers is not in the document]
Comment (sasha-scale) [reply nucleus/__init__.py:331]: I think this is resolved now

File: nucleus/__init__.py [hunk 6]
```diff
@@ -354,9 +363,17 @@ def create_dataset(
         Returns:
             :class:`Dataset`: The newly created Nucleus dataset as an object.
         """
+        warnings.warn(
+            "The default create_dataset('dataset_name', ...) method will be deprecated soon. "
```
Comment (sasha-scale) [inline nucleus/__init__.py:367]:
```suggestion
            "The default create_dataset('dataset_name', ...) method without the is_scene parameter will be deprecated soon in favor of providing the is_scene parameter explicitly. "
```

File: pyproject.toml [hunk 7]
```diff
@@ -21,7 +21,7 @@ exclude = '''
 
 [tool.poetry]
 name = "scale-nucleus"
-version = "0.3.1"
+version = "0.4.0"
```
Comment (sasha-scale) [inline pyproject.toml:24]: we just incremented this for Model CI 😢 
```suggestion
version = "0.4.1"
```

**Review by sasha-scale — APPROVED** [review]
lgtm barring comments

File: tests/test_dataset.py (hunk 1 above)
Comment (pfmark) [reply tests/test_dataset.py:107]: It's both covered in the backend and here now.

**Review by jean-lucas — APPROVED** [review]
LGTM 👍

**Pull request merged**